src/Type/Schema.php

```diff
@@ -410,11 +410,16 @@ private function defaultTypeLoader(string $typeName): ?Type
     }
 
     /**
-     * @param Type|callable():Type $type
+     * @param Type|callable $type
+     *
+     * @template T of Type
+     * @phpstan-param T|callable():T $type
+     * @phpstan-return T
      */
     public static function resolveType($type): Type
     {
         if ($type instanceof Type) {
+            /** @phpstan-var T $type */
```

Comment:
I'd expect phpstan to remember the type information from 

```php
* @param Type|callable $type
* @phpstan-param T|callable():T $type
```

🤔

Reply:
Me too, it seems as though the `instanceof` check erases the generic type. I still think it is better to have the function be well-typed for consumers.